### ISSUE TITLE
shouldShow script error

### ISSUE THREAD

**Issue by bradyy** [Mojito-Fivem/bt-target#24]
Updated to the latest version, function is just a simple inventory check wether if user has the required item.

![image](https://user-images.githubusercontent.com/15060451/126083941-9fed6bc0-5eb9-4973-8d4f-cd12da854ee7.png)

^ Console outputs the function return, which is true

Code below:

![image](https://user-images.githubusercontent.com/15060451/126083953-7586a779-500f-4be1-b20b-411e64a0dc64.png)

Target eye stops working after.

**Comment by LiamDormon** [comment]
Have you tried not embedding other functions in there, I haven't tried that in my testing but just returning true or false works fine.

**Comment by bradyy** [comment]
> Have you tried not embedding other functions in there, I haven't tried that in my testing but just returning true or false works fine.

Yup, returning either true or just false is fine, but embedding a function does not work. (Same for natives)

**Comment by LiamDormon** [comment]
Don't use natives then, run a loop that sets global variables and then you can do
```lua
if condition then
    return true
else
    return false
end
```
 I believe state bags also work.

**Comment by bradyy** [comment]
Actually, a quick fix is to set option.shouldShow to nil after condition checking. Just tried it myself and it is all fine now.

```lua
for _, option in pairs(Zones[_]["targetoptions"]["options"]) do
	if option.shouldShow == nil or option.shouldShow() then
		option.shouldShow = nil
		for _, job in pairs(option.job) do
```

Same for other checks.

**Comment by LiamDormon** [comment]
Submti a PR and I'll review it when I get the time

**Comment by LiamDormon** [comment]
Fixed #25
